### PR TITLE
add nf-test for ucsc/bedclip

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/ucsc/bedclip/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                input[1] = Channel.of([file(params.test_data['sarscov2']['genome']['genome_sizes'], checkIfExists: true)])
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.bedgraph).match("bedgraph") },
+                { assert snapshot(process.out.versions).match("versions") }
```
Comment (mashehu) [inline modules/nf-core/ucsc/bedclip/tests/main.nf.test:30]:
```suggestion
                { assert snapshot(process.out).match() }
```
That would be more concise I think

**Review by mashehu — APPROVED** [review]

**Comment by SPPearce** [comment]
Why are we not removing the pytest files?

**Comment by mashehu** [comment]
> Why are we not removing the pytest files?

We just decided to do that now. Docs update on the way (https://github.com/nf-core/website/pull/2119), linting as well (https://github.com/nf-core/tools/pull/2531)

File: modules/nf-core/ucsc/bedclip/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    name "Test Process UCSC_BEDCLIP"
+    script "../main.nf"
+    process "UCSC_BEDCLIP"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "ucsc"
+    tag "ucsc/bedclip"
+
+    test("sarscov2") {
+        config "./nextflow.config"
```
Comment (sateeshperi) [inline modules/nf-core/ucsc/bedclip/tests/main.nf.test:12]: move this to L6

File: modules/nf-core/ucsc/bedclip/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    name "Test Process UCSC_BEDCLIP"
+    script "../main.nf"
+    process "UCSC_BEDCLIP"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "ucsc"
+    tag "ucsc/bedclip"
+
+    test("sarscov2") {
```
Comment (sateeshperi) [inline modules/nf-core/ucsc/bedclip/tests/main.nf.test:11]:
```suggestion
    test("sarscov2 [bedgraph]") {
```

**Pull request merged**